src/collect/collector.rs

```diff
@@ -134,8 +136,35 @@ impl Collectors {
     /// Setup user defined input filter.
     fn setup_filters(probes: &mut ProbeManager, collect: &Collect) -> Result<()> {
         if let Some(f) = &collect.args()?.packet_filter {
-            let fb = FilterPacket::from_string(f.to_string())?;
-            probes.register_filter(Filter::Packet(BpfFilter(fb.to_bytes()?)))?;
+            // L2 filter MUST always succeed. Any failure means we need to bail.
+            let fb = FilterPacket::from_string_opt(f.to_string(), FilterPacketType::L2)?;
+
+            probes.register_filter(Filter::Packet(BpfFilter {
+                key: FilterPacketType::L2,
+                data: fb.to_bytes()?,
+            }))?;
+
+            // L3 filter is non mandatory.
+            let fb = if f.contains("ether[") {
+                debug!("skipping L3 filter generation (ether[n:m] not allowed)");
+                FilterPacket::empty_filter()
+            } else {
+                match FilterPacket::from_string_opt(f.to_string(), FilterPacketType::L3) {
+                    Err(e) => {
+                        debug!("skipping L3 filter generation ({e}).");
+                        FilterPacket::empty_filter()
+                    }
+                    Ok(f) => {
+                        debug!("L3 filter generated.");
+                        f
+                    }
+                }
+            };
+
+            probes.register_filter(Filter::Packet(BpfFilter {
+                key: FilterPacketType::L3,
+                data: fb.to_bytes()?,
+            }))?;
```

Comment:
ether[] it's already there, but in general, all the times we skip L3 we log the reason.
L2 is mandatory, so I found that logging it every time is kind of redundant.
They are all under DEBUG log level, though. Do you suggest moving it to INFO?

Reply:
IMHO, if we keep it as `DEBUG` it should be functional for debugging, and avoiding L2 is fine (because it `bail!()` in that case).
If we want `INFO` we should probably go with `L2[+L3] installed`. I tend to prefer `debug!()` but I have no strong preference. Whatever you prefer is fine.

Reply:
SGTM, I'm convinced that it's probably better to make "automatic logic" stand out. The user would align the expectations in this way. The reason (debug) is useful anyway, mostly for us to quickly understand what's going on.

While we're speaking, could make sense to dump `mac_header` and `network_header` in the skb module?
Maybe as part of a debug section?
Or consider some raw reporting of `bpf_snprintf_btf()`, e.g. tackling the local trace pipe and making that info available at some log level.